charts/kubermatic-operator/crd/k8c.io/kubermatic.k8c.io_clusters.yaml

```diff
@@ -1472,6 +1472,11 @@ spec:
               debugLog:
                 description: DebugLog enables more verbose logging by KKP's usercluster-controller-manager.
                 type: boolean
+              enableKubernetesDashboard:
+                default: true
+                description: EnableKubernetesDashboard controls whether the kubernetes-dashboard
+                  should be deployed for the user cluster or not.
+                type: boolean
```

Comment:
I know that we have the exact same flag for OSM, but during the 2.20 restructing we kind of decided to not have these boolean flags anymore, because they are impossible to extend. If another dashboard-related option is added in the future, we end up with multiple flags next to each other, like here: https://github.com/kubermatic/kubermatic/blob/master/pkg/apis/kubermatic/v1/settings.go#L55-L58

I would propose that we use a more extensible structure like

```yaml
spec:
  kubernetesDashboard:
    enabled: true|false
```